### PR TITLE
Return appropriate error codes

### PR DESCRIPTION
**Pull request by jkakavas**
When we fail to validate an Authentication request because the
request is invalid, respond with 400 instead of 500.

Tests for these are in the WIP https://github.com/elastic/elasticsearch/pull/53130

### CODE REVIEW

**Comment by jkakavas** [comment]
@elasticmachine update branch

File: x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/action/TransportSamlInitiateSingleSignOnAction.java [hunk 1]
```diff
@@ -64,7 +66,8 @@ protected void doExecute(Task task, SamlInitiateSingleSignOnRequest request,
                     }
                     final SecondaryAuthentication secondaryAuthentication = SecondaryAuthentication.readFromContext(securityContext);
                     if (secondaryAuthentication == null) {
-                        listener.onFailure(new IllegalStateException("Request is missing secondary authentication"));
+                        listener.onFailure(
+                            new ElasticsearchStatusException("Request is missing secondary authentication", RestStatus.FORBIDDEN));
```
Comment (tvernum) [inline x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/action/TransportSamlInitiateSingleSignOnAction.java]: Can we make it `ElasticsearchSecurityException` instead?

```suggestion
                            new ElasticsearchSecurityException("Request is missing secondary authentication", RestStatus.FORBIDDEN));
```

**Review by tvernum — APPROVED** [review]
LGTM, but I'd prefer ElasticsearchSecurityException

**Comment by jkakavas** [comment]
@elasticmachine run elasticsearch-ci/1

**Pull request merged**